docs/configuration.rst

```diff
@@ -198,23 +198,31 @@ An example:
 
 Depending on your particular deployment you most probably will want to change the default.
 
-.. _advanced_cfg:
+.. _advanced_settings:
 
 Advanced settings
 -----------------
 
-The ``CREDENTIAL_ROUNDS`` (default: 12) setting is the number of rounds used by the 
-password hashing scheme. The number of rounds can be reduced in case faster 
-authentication is needed or increased when additional protection is desired. 
-Keep in mind that this is a mitigation against offline attacks on password hashes, 
+The ``API`` (default: False) setting controls if the API endpoint is reachable.
+
+The ``WEB_API`` (default: /api) setting configures the endpoint that the API
+listens on publicly&interally. The path must always start with a leading slash.
+
+The ``API_TOKEN`` (default: None) enables the API endpoint. This token must be
+passed as request header to the API as authentication token.
```

Comment:
I think it is a little bit unclear (and maybe redundant) waht the API, WEB_API and API_TOKEN variables do.

Do I understand correctly, that
1. by setting API_TOKEN only: I enable the API internally on /api
2. by setting WEB_API only: nothing happens
3. by setting API only: nothing happens
4. by setting API and API_TOKEN: the api is exposed to the world (path can be changed by WEB_API)

I think we should describe this clearer.

Reply:
What is not clear about "The API_TOKEN (default: None) `enables the API endpoint.`?"

All three settings are required. Some people will want the option to make it publicly exposed or not. some people will want to change the end point (/api), just like the other end points like WEB_ADMIN. For this first version the API token is static. In a future PR we could explore configuring the API token per user via the web admin.

by setting API_TOKEN only: I enable the API internally on /api
=>
That is correct.

by setting WEB_API only: nothing happens
=>
That is correct.

by setting API only: nothing happens
=> 
That is correct.

by setting API and API_TOKEN: the api is exposed to the world (path can be changed by WEB_API)
=>
That is correct.

I could add a paragraph above these 3 settings that states that `API` and `API_TOKEN`  should be both configured for using the API. I could also add this to the `docs/api.rst`

Reply:
If I may, I think the naming could be better:

API -> API_ENABLED
WEB_API -> API_PATH
API_TOKEN -> is ok as is. 

??? 

Scott

Reply:
No. I used the same naming conventions as the other components. I don't want to mix naming schemes. 
ADMIN also uses `ADMIN` for enabling (exposing publicly) and `WEB_ADMIN` for the endpoint.
WEBMAIL also uses `WEBMAIL` for enabling (exposing publicly) and `WEB_WEBMAIL` for the endpoint.



Maybe configuring the API to be publicly reachable (`API`), should be moved from the configuration reference to the setup utility. We remove `API` from the documentation. This is how we currently do it for ADMIN & WEBMAIL. This makes it more consistent.

Reply:
I fear we will have complaints about setting WEB_API and the api remaining on "/api" internally.
And complaints about setting API to true and "nothing happens".
I know Mailu uses this kind of settings in other places, but we could start making it more accessible.

I think we should not put API into the setup utility, as it is not needed there.
Only variables changing the compose-file need to be in the setup utility.
Everything else can be changed "on the fly" - and maybe at some time in the admin interface, too.

Reply:
That is not how the setup utility is currently designed. Many settings are for the mailu.env file only. The goal of the setup utility is to enable less experienced users to configure a new mailu installation. Ideally users new have to touch mailu.env.

>I fear we will have complaints about setting WEB_API and the api remaining on "/api" internally.
>And complaints about setting API to true and "nothing happens".
>I know Mailu uses this kind of settings in other places, but we could start making it more accessible.

Using the setup utility will prevent this. In the setup utility we will enforce that you cannot provide a non-empty value for API_TOKEN and WEB_API, if the API is enabled. Just like when you enable ADMIN or WEBMAIL in the setup utility.

Modifying the mailu.env file is scary and error prone for many of our users. 

>I fear we will have complaints about setting WEB_API and the api remaining on "/api" internally.

I do not agree. This is in line with all our internal components. They all have static endpoints defined. It is by design that you cannot change the internal endpoints (/ui, /sso, /internal, /api, etc).

Reply:
> I do not agree. This is in line with all our internal components.

Hi @Diman0 - I never said this is not "in line" - other setup parameters can also be hard to understand.

But it's okay for me to configure it like this. Maybe we will have a big config-change later where we eill separate the things which can be configured at runtime from those which can not. Things like the api for example could be (de)activated at runtime and also the token could be changed live in the admin interface.